internal/gob/gob.go

```diff
@@ -0,0 +1,36 @@
+package gob
+
+import (
+	"encoding/gob"
+	"time"
+
+	dynamotypes "github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/jackc/pgx/v5/pgtype"
+	neosynctypes "github.com/nucleuscloud/neosync/internal/neosync-types"
+)
+
+// need to register all the types that are used in the connection data service
+// because we use interfaces for the types
+func RegisterGobTypes() {
+	gob.Register(map[string]any{})
+	gob.Register(map[string]any{})
+	gob.Register(pgtype.Numeric{}) // TODO fix this
+	gob.Register([]any{})
+	gob.Register(time.Time{})
```

Comment:
interesting to see native go types being registered here. that is very strange that they arent natively supported by the language